### PR TITLE
Yohan/raz 366 migrate oracle node to GitHub actions

### PR DESCRIPTION
**Pull request by yohanelly95**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue) 
- Migration to Github Actions
- Removal of circle ci dependancy

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by ashish10677 — APPROVED** [review]

**Review by Yashk767 — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: "18"
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17.6
```
Comment (SkandaBhat) [inline .github/workflows/ci.yml:28]: update go-version to 1.21

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        run: >
+          sudo add-apt-repository -y ppa:ethereum/ethereum
+
+          sudo apt-get update
+
+          sudo apt-get install -y npm ethereum
+
+          npm install
+
+          go get -d github.com/ethereum/go-ethereum@v1.10.8 \
```
Comment (SkandaBhat) [inline .github/workflows/ci.yml:39]: update go ethereum 1.12.2

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+          && go install github.com/ethereum/go-ethereum/cmd/abigen@v1.10.8 \
```
Comment (SkandaBhat) [inline .github/workflows/ci.yml:41]: update abigen to 1.12.2

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+          && go install github.com/mattn/goveralls@v0.0.11 \
+
+          && go install github.com/ory/go-acc@v0.2.7 \
+
+          && go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.46.2
```
Comment (SkandaBhat) [inline .github/workflows/ci.yml:47]: update go lint 1.54.2

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+        run: >
+          sudo add-apt-repository -y ppa:ethereum/ethereum
+
+          sudo apt-get update
+
+          sudo apt-get install -y npm ethereum
+
+          npm install
+
+          go get -d github.com/ethereum/go-ethereum@v1.10.8 \
```
Comment (SkandaBhat) [inline .github/workflows/ci.yml:102]: do the same here

**Pull request closed without merging**